### DISCUSSION TITLE
How is \\wsl.localhost implemented?

### DISCUSSION THREAD

**Discussion by fiddyschmitt** [microsoft/WSL#9043] [General]
Hi,

The `\\wsl.localhost\` share is a nice way of accessing folders within WSL. Could you please let me know how it was implemented? I normally use [Dokan](https://github.com/dokan-dev/dokany) to implement Virtual File Systems, so I'd like to understand how the `\\wsl.localhost\` share works (preferably in C#).

Thanks!
Fidel

**Comment by BarsMonster** [comment]
My understanding is that's 9P in both directions (windows->linux and linux->windows), if it's WSL2

**Comment by anushhkumar** [comment]
I did the exactly same steps with you and got the same [result](https://forum.benchmark.pl/topic/282697-problem-z-geforce-experience-b%C5%82%C4%85d-0x0003/).
